platforms/hyperledger-besu/charts/generate_ambassador_certs/templates/job.yaml

```diff
@@ -67,57 +71,35 @@ spec:
           value: {{ $.Values.vault.role }}
         - name: NODE_NAME
           value: {{ $.Values.node.name }}
+        - name: VAULT_TYPE
+          value: "{{ $.Values.vault.type }}"
         - name: MOUNT_PATH
           value: "/certificates"
         command: ["sh", "-c"]
         args:
         - |-
-
-          validateVaultResponse () {
-            if echo ${2} | grep "errors"; then
-              echo "ERROR: unable to retrieve ${1}: ${2}"
-              exit 1
-            fi
-            if  [ "$3" == "LOOKUPSECRETRESPONSE" ]
-            then
-              http_code=$(curl -sS -o /dev/null -w "%{http_code}" \
-              --header "X-Vault-Token: ${VAULT_CLIENT_TOKEN}" \
-              ${VAULT_ADDR}/v1/${vault_secret_key})
-              curl_response=$?
-              if test "$http_code" != "200" ; then
-                  echo "Http response code from Vault - $http_code"
-                  if test "$curl_response" != "0"; then
-                     echo "Error: curl command failed with error code - $curl_response"
-                     exit 1
-                  fi
-              fi
-            fi
-          }
-
-          KUBE_SA_TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
+          source /scripts/bevel-vault.sh
+          echo "Getting the vault Token.."
+          vaultBevelFunc "init"
           echo "Getting secrets from Vault Server: ${VAULT_ADDR}"
           #Login to Vault and so I can get an approle token
-          VAULT_CLIENT_TOKEN=$(curl -sS --request POST ${VAULT_ADDR}/v1/auth/${KUBERNETES_AUTH_PATH}/login \
-            -H "Content-Type: application/json" \
-            -d '{"role":"'"${VAULT_APP_ROLE}"'","jwt":"'"${KUBE_SA_TOKEN}"'"}' | \
-            jq -r 'if .errors then . else .auth.client_token end')
-          validateVaultResponse 'vault login token' "${VAULT_CLIENT_TOKEN}"
+          validateVaultResponseHashicorp 'vault login token' "${VAULT_TOKEN}"
 
           OUTPUT_PATH=${MOUNT_PATH}/check_certs;
           mkdir -p ${OUTPUT_PATH}
 
-          vault_secret_key=${VAULT_SECRET_ROOTCA}
+          vault_secret_key="${VAULT_SECRET_ENGINE}/${VAULT_SECRET_ROOTCA}"
           echo "Checking certs in ${vault_secret_key}"
-          LOOKUP_SECRET_RESPONSE=$(curl -sS --header "X-Vault-Token: ${VAULT_CLIENT_TOKEN}" ${VAULT_ADDR}/v1/${vault_secret_key} | jq -r 'if .errors then . else . end')
-          CA_PEM=$(echo ${LOOKUP_SECRET_RESPONSE} | jq -r '.data.data["rootca_pem"]' 2>&1)
-          CA_KEY=$(echo ${LOOKUP_SECRET_RESPONSE} | jq -r '.data.data["rootca_key"]' 2>&1)
+          vaultBevelFunc "readJson" ${vault_secret_key} 
+          CA_PEM=$(echo ${VAULT_SECRET} | jq -r '.data["rootca_pem"]' 2>&1)
```

Comment:
which version of vault is this tested on?

Reply:
Its tested on 1.7.1

Reply:
the data.data is required, but its already sending the .data from the script
VAULT_SECRET=$(curl --header "X-Vault-Token: ${VAULT_TOKEN}" ${VAULT_ADDR}/v1/${1} | jq -r 'if .errors then "null" else .data end')